src/plugins/validation/oas3/semantic-validators/pagination-ibm.js

```diff
@@ -216,3 +219,16 @@ module.exports.validate = function({ resolvedSpec }, config) {
 
   return messages;
 };
+
+function mergeAllOfSchemaProperties(schema) {
+  const mergedSchema = {};
+  const allOfArr = schema['allOf'];
+  if (allOfArr && Array.isArray(allOfArr)) {
+    // delete now, so it not merged back into the final result
+    delete schema['allOf'];
+    allOfArr.forEach(function(allOfSchema) {
+      merge(mergedSchema, allOfSchema);
```

Comment:
How does this handle property conflicts? For example, each `allOfSchema` will likely have a field called `properties` - will they be overwritten or are subfields merged as well? It may be a good idea to add one more schema to the `allOf` list in your test just to verify this

Reply:
The `lodash` merge function does do a deep merge.

I will add a test to ensure it is actually deep merge and to ensure we preserve the deep merge behavior in future updates.

Reply:
Although `lodash` does do a deep merge, I may need to consider some other possible consequences here.

Reply:
I did two things to follow up on this:

1. I made the merging function a utility function and wrote unit tests separately, which is a much better approach than adding it to the module and only having acceptance tests.
2. I added another schema to the allOf array to ensure the properties are not being overwritten. The `pagination-ibm` module uses the `required` and `properties` fields from the schema, and although it's unclear how to merge things like `description` and `summary`, the tests ensure `required` and `properties` are merged properly.